R/frosting.R

```diff
@@ -87,8 +93,83 @@ validate_has_postprocessor <- function(x, ..., call = caller_env()) {
   invisible(x)
 }
 
+#' @rdname add_frosting
+#' @export
+update_frosting <- function(x, frosting, ...) {
+  x <- remove_frosting(x)
+  add_frosting(x, frosting)
+}
+
+
+#' Adjust a layer in an `epi_workflow` or `frosting`
+#'
+#' Make a parameter adjustment to a layer in either an
+#' `epi_workflow` or `frosting` object.
+#'
+#'
+#' @details This function can either adjust a layer in a `frosting` object
+#' or a layer from a `frosting` object in an `epi_workflow`. In any case, the
+#' argument name and update value must be inputted as `...`.
+#' See the examples below for brief illustrations of both types of updates.
+#'
+#' @param x An `epi_workflow` or `frosting` object
+#'
+#' @param layer_num the number of the layer to adjust
+#'
+#' @param ... Used to input a parameter adjustment
+#'
+#' @return
+#' `x`, updated with the adjustment to the specified `frosting` layer.
+#'
+#' @export
+#' @examples
+#' jhu <- case_death_rate_subset %>%
+#'   filter(time_value > "2021-11-01", geo_value %in% c("ak", "ca", "ny"))
+#' r <- epi_recipe(jhu) %>%
+#'   step_epi_lag(death_rate, lag = c(0, 7, 14)) %>%
+#'   step_epi_ahead(death_rate, ahead = 7) %>%
+#'   step_epi_naomit()
+#'
+#' wf <- epi_workflow(r, parsnip::linear_reg()) %>% fit(jhu)
+#'
+#' # in the frosting from the workflow
+#' f1 <- frosting() %>% layer_predict() %>% layer_threshold(.pred)
+#'
+#' wf2 = wf %>% add_frosting(f1)
+#'
+#' # Adjust `layer_threshold` to have an upper bound of 1
+#' # in the `epi_workflow`
+#' wf2 = wf2 %>% adjust_frosting(layer_num = 2, upper = 1)
+#' extract_frosting(wf2)
+#'
+#' # Adjust `layer_threshold` to have an upper bound of 1
+#' # in the `frosting` object
+#' f2 = f1 %>% adjust_frosting(layer_num = 2, upper = 5)
+#' extract_frosting(wf2)
+#'
+adjust_frosting <- function(x, layer_num, ...) {
+  UseMethod("adjust_frosting")
+}
+
+#' @rdname adjust_frosting
+#' @export
+adjust_frosting.epi_workflow <- function(
```

Comment:
Same idea with adjusting based on names (as well as numbers) as above.

Probably the best way is to replace `layer_num` with `which_layer`. If it's numeric, then great. If it's character, then check for uniqueness.